addons/website_sale/views/website_pages_views.xml

```diff
@@ -31,13 +31,36 @@
             </field>
             <xpath expr="//tree">
                 <field name="is_seo_optimized"/>
-                <field name="is_published"/>
-
+                <field name="scheduled" column_invisible="1"/>
+                <field name="is_published" invisible="scheduled and is_published"/>
+                <field name="publish_at" string="Scheduled" invisible="not scheduled"/>
                 <field name="website_id" position="move"/>
             </xpath>
         </field>
     </record>
-
+    <record id="product_pages_form_view" model="ir.ui.view">
+        <field name="name">Product pages form view</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_only_form_view"/>
+        <field name="arch" type="xml">
+            <notebook position="inside">
+                <page name="publishing" string="Publish">
+                    <group>
+                        <field name="scheduled" invisible="1"/>
+                        <field string="Published" name="is_published" widget="boolean_toggle" readonly="scheduled"></field>
+                        <label invisible="is_published" for="publish_at" string="Publish at"/>
+                        <div invisible="is_published" name="publish_at_div" class="o_row">
+                            <field name="publish_at" widget="datetime" options="{'min_date': datetime.datetime.now().to_utc().strftime('%Y-%m-%d %H:%M:%S')}"/>
+                            <button name="action_unschedule" type="object" invisible="not publish_at">Unschedule</button>
+                        </div>
+                    </group>
+                    <div class="alert alert-info w-100" invisible="not scheduled" role="alert">
+                        <small>This product will automatically go live on the specified date</small>
+                    </div>
+                </page>
+            </notebook>
```

Comment:
Isn't there a way to share a QWeb XML structure to reuse it in different models and just change what needs to be changed? Having this 4 times in 4 different files doesn't look optimal, but maybe there isn't a better solution. :shrug: